Ryujinx.HLE/Input/Hid.cs

```diff
@@ -0,0 +1,145 @@
+                     (NpadColor.Buttons_Neon_Blue, NpadColor.Buttons_Neon_Blue));
+            }
+
+            PrimaryController.Connect(ControllerId);
+        }
+
+        private HidControllerButtons UpdateStickButtons(
+            HidJoystickPosition LeftStick,
+            HidJoystickPosition RightStick)
+        {
+            HidControllerButtons Result = 0;
+
+            if (RightStick.DX < 0)
+            {
+                Result |= HidControllerButtons.RStickLeft;
+            }
+
+            if (RightStick.DX > 0)
+            {
+                Result |= HidControllerButtons.RStickRight;
+            }
+
+            if (RightStick.DY < 0)
+            {
+                Result |= HidControllerButtons.RStickDown;
+            }
+
+            if (RightStick.DY > 0)
+            {
+                Result |= HidControllerButtons.RStickUp;
+            }
+
+            if (LeftStick.DX < 0)
+            {
+                Result |= HidControllerButtons.LStickLeft;
+            }
+
+            if (LeftStick.DX > 0)
+            {
+                Result |= HidControllerButtons.LStickRight;
+            }
+
+            if (LeftStick.DY < 0)
+            {
+                Result |= HidControllerButtons.LStickDown;
+            }
+
+            if (LeftStick.DY > 0)
+            {
+                Result |= HidControllerButtons.LStickUp;
+            }
+
+            return Result;
+        }
+
+        public void SetTouchPoints(params HidTouchPoint[] Points)
+        {
+            long TouchScreenOffset = HidPosition + HidTouchScreenOffset;
+
+            long LastEntry = Device.Memory.ReadInt64(TouchScreenOffset + 0x10);
+
+            long CurrEntry = (LastEntry + 1) % HidEntryCount;
+
+            long Timestamp = GetTimestamp();
+
+            Device.Memory.WriteInt64(TouchScreenOffset + 0x0,  Timestamp);
+            Device.Memory.WriteInt64(TouchScreenOffset + 0x8,  HidEntryCount);
+            Device.Memory.WriteInt64(TouchScreenOffset + 0x10, CurrEntry);
+            Device.Memory.WriteInt64(TouchScreenOffset + 0x18, HidEntryCount - 1);
+            Device.Memory.WriteInt64(TouchScreenOffset + 0x20, Timestamp);
+
+            long TouchEntryOffset = TouchScreenOffset + HidTouchHeaderSize;
+
+            long LastEntryOffset = TouchEntryOffset + LastEntry * HidTouchEntrySize;
+
+            long SampleCounter = Device.Memory.ReadInt64(LastEntryOffset) + 1;
+
+            TouchEntryOffset += CurrEntry * HidTouchEntrySize;
+
+            Device.Memory.WriteInt64(TouchEntryOffset + 0x0, SampleCounter);
+            Device.Memory.WriteInt64(TouchEntryOffset + 0x8, Points.Length);
+
+            TouchEntryOffset += HidTouchEntryHeaderSize;
+
+            const int Padding = 0;
+
+            int Index = 0;
+
+            foreach (HidTouchPoint Point in Points)
+            {
+                Device.Memory.WriteInt64(TouchEntryOffset + 0x0,  SampleCounter);
```

Comment:
I usually use long-form hex to avoid confusion. Also has the benefit of aligning these with the longer hex values below.

```suggestion
                Device.Memory.WriteInt64(TouchEntryOffset + 0x00,  SampleCounter);
```